x-pack/solutions/observability/plugins/observability/public/pages/rule_details/components/header_actions.tsx

```diff
@@ -15,21 +15,49 @@ import {
   EuiText,
 } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
-
+import { noop } from 'lodash';
+import { useFetchRule } from '../../../hooks/use_fetch_rule';
+import { useKibana } from '../../../utils/kibana_react';
+import { useEnableRule } from '../../../hooks/use_enable_rule';
+import { useDisableRule } from '../../../hooks/use_disable_rule';
 interface HeaderActionsProps {
+  ruleId: string;
   isLoading: boolean;
   isRuleEditable: boolean;
   onDeleteRule: () => void;
   onEditRule: () => void;
 }
 
 export function HeaderActions({
+  ruleId,
   isLoading,
   isRuleEditable,
   onDeleteRule,
   onEditRule,
 }: HeaderActionsProps) {
+  const { services } = useKibana();
+  const {
+    triggersActionsUi: {
+      getRuleSnoozeModal: RuleSnoozeModal,
+      getUntrackModal: UntrackAlertsModal,
+      getRuleHelpers,
+    },
+  } = services;
+
   const [isRuleEditPopoverOpen, setIsRuleEditPopoverOpen] = useState(false);
+  const [snoozeModalOpen, setSnoozeModalOpen] = useState<boolean>(false);
+  const [isUntrackAlertsModalOpen, setIsUntrackAlertsModalOpen] = useState<boolean>(false);
+
+  const { mutateAsync: enableRule } = useEnableRule();
+  const { mutateAsync: disableRule } = useDisableRule();
+
+  const onDisableModalClose = () => {
+    setIsUntrackAlertsModalOpen(false);
+  };
+
+  const onDisableModalOpen = () => {
+    setIsUntrackAlertsModalOpen(true);
+  };
```

Comment:
```suggestion
```

Another opinionated commented; I don't think declaring these functions is necessary.